### ISSUE TITLE
Issue with Invoke-IcingaCheckMSSQLBackupStatus on a single machine

### ISSUE THREAD

**Issue by iiM-Dawnbreaker** [Icinga/icinga-powershell-mssql#35]
### Expected Behavior
`[OK] MSSQL Backup (MSSQLSERVER)
| 'average_size'=3363200B;; 'log_age'=0s;; 'size'=80716800B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'avera
ge_size'=2900267B;; 'log_age'=0s;; 'size'=69606400B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'
=18624810000B;; 'log_age'=0s;; 'size'=446995300000B;; 'status'=0;6;5 'execution_time'=60s;; 'age'=46800s;; 'size'=420354
80000B;; 'execution_time'=60s;; 'age'=1688400s;; 'average_size'=42035480000B;; 'log_age'=0s;; 'average_size'=3379200B;;
'log_age'=0s;; 'size'=81100800B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=132823000B;; 'log_a
ge'=0s;; 'size'=3187753000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=21898940000B;; 'log_age
'=0s;; 'size'=525574600000B;; 'status'=0;6;5 'execution_time'=60s;; 'age'=46800s;; 'average_size'=331950200B;; 'log_age'
=0s;; 'size'=7966804000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=14316800B;; 'log_age'=0s;;
 'size'=343603200B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=5433002B;; 'log_age'=0s;; 'size'
=130392100B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=313211800B;; 'log_age'=0s;; 'size'=4698
178000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=23404900000B;; 'log_age'=0s;; 'size'=351073
500000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=35883010B;; 'log_age'=0s;; 'size'=861192200
B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=2848131000B;; 'log_age'=0s;; 'size'=68355140000B;
; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=4113067B;; 'log_age'=0s;; 'size'=98713600B;; 'statu
s'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=583269400B;; 'log_age'=0s;; 'size'=13998470000B;; 'status'=
0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=75516800B;; 'log_age'=0s;; 'size'=1812403000B;; 'status'=0;6;5
 'execution_time'=0s;; 'age'=46800s;; 'average_size'=6646400B;; 'log_age'=0s;; 'size'=159513600B;; 'status'=0;6;5 'execu
tion_time'=0s;; 'age'=46800s;; 'average_size'=305439500B;; 'log_age'=0s;; 'size'=7330548000B;; 'status'=0;6;5 'execution
_time'=0s;; 'age'=46800s;; 'average_size'=3384533B;; 'log_age'=0s;; 'size'=81228800B;; 'status'=0;6;5 'execution_time'=0
s;; 'age'=46800s;; 'average_size'=58740250000B;; 'log_age'=0s;; 'size'=1409766000000B;; 'status'=0;6;5 'execution_time'=
240s;; 'age'=46800s;; 'average_size'=43882670B;; 'log_age'=0s;; 'size'=1053184000B;; 'status'=0;6;5 'execution_time'=0s;
; 'age'=46800s;; 'average_size'=59584680B;; 'log_age'=0s;; 'size'=1430032000B;; 'status'=0;6;5 'execution_time'=0s;; 'ag
e'=46800s;; 'average_size'=25255470B;; 'log_age'=0s;; 'size'=606131200B;; 'status'=0;6;5 'execution_time'=60s;; 'age'=46
800s;; 'average_size'=2151852000B;; 'log_age'=0s;; 'size'=51644440000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=4680
0s;; 'average_size'=61355730B;; 'log_age'=0s;; 'size'=1472538000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;;
'average_size'=125545900B;; 'log_age'=0s;; 'size'=3013101000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'ave
rage_size'=173949800B;; 'log_age'=0s;; 'size'=4174795000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average
_size'=32935470B;; 'log_age'=0s;; 'size'=790451200B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'
=624253800B;; 'log_age'=0s;; 'size'=14982090000B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;; 'average_size'=29
44939B;; 'log_age'=0s;; 'size'=70678530B;; 'status'=0;6;5 'execution_time'=0s;; 'age'=46800s;;
0`

### Current Behavior
First starts off with an Exception before putting out the Request resulting in Icinga Director changing into Unknown with the Exception Message
```
PS Z:\> Invoke-IcingaCheckMSSQLBackupStatus -SqlUsername someuser -SqlPassword (ConvertTo-IcingaSecureString somepassword)
Ausnahme beim Aufrufen von "WarnIfMatch" mit 1 Argument(en):  "Ausnahme beim Aufrufen von "WarnIfLike" mit 1
Argument(en):  "Ausnahme beim Aufrufen von "__SetCheckState" mit 2 Argument(en):  "Ausnahme beim Aufrufen von
"__SetCheckOutput" mit 0 Argument(en):  "Ausnahme beim Aufrufen von "__SetPerformanceData" mit 0 Argument(en):  """
kann nicht verglichen werden, da er nicht "IComparable" ist."""""
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-mssql\plugins\Invoke-IcingaCheckMSSQLBackupStatus.psm1:155 Zeichen:9
+         $DBPackage.AddCheck(
+         ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ScriptMethodRuntimeException

[OK] MSSQL Backup (MSSQLSERVER)
| 'average_size'=3363200B;; 'log_age'=0s;; 'size'=80716800B;; 'status'=0;6;5 'execution_time'=0s;;
```

Possible Solution
N/A

### Steps to Reproduce (for bugs)
Configuring the Service in Icinga Director and pushing it out onto a Database Client with MSSQL installed.
Running the Command only with Sql Username and Password and Threshholds for BackupAge
`Invoke-IcingaCheckMSSQLBackupStatus Arguments: -LastBackupAgeWarning 72h -LastBackupAgeCritical 168h -IncludeDatabase System.Object[] -SqlUsername someuser -SqlPassword System.Security.SecureString`

### Your Environment
PowerShell Version used ($PSVersionTable.PSVersion): 5.1.14409.1005
MS SQL Server: 2019
Operating System and version (Get-IcingaWindowsInformation Win32_OperatingSystem | Select-Object Version, BuildNumber, Caption):  6.3.9600 | 9600 | Microsoft Windows Server 2012 R2 Datacenter
PowerShell Framework Version: 1.5.2
PowerShell MSSQL Plugin Version: 1.2.0
PowerShell Plugin Version: 1.5.1

The Server itself is on the sam Updatelevel etc. as the other Server.

**Comment by 408573** [comment]
Same problem here. Are there any ideas how to fix this? 
Thanks a lot!

**Comment by Br0cken** [comment]
We also ran into this issue. For debugging you can manually execute the SELECT query from _Get-IcingaMSSQLBackupOverallStatus.psm1_ in SQL Server Management Studio
It returned us a long list with many deprecated entrys. 

To clean this up we set up a maintanance plan to cleanup the backup and restore history for entrys older than 4 weeks. After the  cleanup job finished, the check executed just fine. 
Positive side effect: Check execution time decreased significantly.

**Comment by LordHepipud** [comment]
Thank you for the issue and please excuse the long wait time. Atleast it has been not a year 😕

Version 1.4.0 (releasing on Tuesday, 30th August), will fix the backup handling by allowing to set on how many days should be included: #42

I assume this will fix this issue?
